A65-xds-mtls-creds-in-bootstrap.md

```diff
@@ -0,0 +1,97 @@
+A65: mTLS Credentials in xDS Bootstrap File
+----
+* Author(s): @markdroth
+* Approver: @ejona86
+* Status: {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2023-05-24
+* Discussion at: https://groups.google.com/g/grpc-io/c/h_LQnTimyt4
+
+## Abstract
+
+This proposal adds support for configuring the use of mTLS for
+communicating with the xDS server via the xDS bootstrap file.
+
+## Background
+
+[gRFC A27][A27] defines the xDS bootstrap file format for setting
+the channel creds used to communicate with the xDS server, with the
+initial options of `google_default` and `insecure`.  It suggested that a
+general-purpose mechanism could be added to configure arbitrary channel
+creds types, which has subsequently been done, albeit not intended as
+a public API.
+
+[gRFC A29][A29] describes the certificate provider framework used for
+configuring mTLS for data plane communication in xDS.  It also defines
+the file-watcher certificate provider.
+
+This proposal defines a new channel creds type to be used in the
+bootstrap file for using mTLS with the xDS server, leveraging the
+functionality of the file-watcher certificate provider.
+
+### Related Proposals: 
+* [gRFC A27: xDS-Based Global Load Balancing][A27]
+* [gRFC A29: xDS-Based Security for gRPC Clients and Servers][A29]
+
+## Proposal
+
+We will define a new credential type in the bootstrap file called `tls`.
+Its configuration will be essentially the same as that of the file-watcher
+certificate provider described in [gRFC A29][A29].  Specifically, the
+config will look like this:
+
+```json
+{
+  // Path to identity certificate file.
+  // If set, mTLS will be used; if unset, normal TLS will be used.
+  "certificate_file": <string>,
+
+  // Paths to private key and CA certificate files.
+  // If either of these fields are set, both must be set.
```

Comment:
These shouldn't be paired together. `certificate_file` is the public certificate chain for the private key `private_key_file`. Those two should be paired together. `ca_certificate_file` is the trust root for the remote's certificate.

Reply:
Doh!  Yeah, I somehow got these totally conflated in my head when I wrote this.  Thanks for catching this.

Fixed.